Added EmptyState for transactions in AssetScene

gemdev111 opened the pull request:
Refactored AssetScene
Removed unused action, minor adjustments
Added empty State if no transactions available #75 
<table>
  <tr>
    <td><img src="https://github.com/gemwalletcom/gem-ios/assets/171273137/5602fc3e-f1ec-4154-9ee2-9739930bbea4" style="width: 45%;" /></td>
    <td><img src="https://github.com/gemwalletcom/gem-ios/assets/171273137/59fa5e5a-b252-4659-86f5-baf069d54a2d" style="width: 45%;" /></td>
  </tr>
</table>

gemdev111 committed:
Minor refactored AssetScene
Removed unused action, make private properties
Added empty State if no transactions available

gemcoder21 commented on Gem/Asset/AssetScene.swift:
```diff
@@ -168,36 +151,67 @@ struct AssetScene: View {
+extension AssetScene {
+    private func fetch() {
+        Task {
+            do {
```
can you wrap this method into view model? so you just call `model.update(wallet)`

gemdev111 committed:
Minor updates ref to @gemcoder21 review
Moved services(requests) logic to vm
Added several todos to handle errors in vm
Added access level control to vm, found several unused methods ( marked them unused for now )

gemcoder21 commented on Gem/Assets/ViewModels/AssetSceneViewModel.swift:
```diff
@@ -162,28 +102,89 @@ class AssetSceneViewModel: ObservableObject {
+        TransactionsRequest(walletId: walletModel.wallet.id, type: .asset(assetId: assetModel.asset.id), limit: transactionsLimit)
+    }
+
+    // currently not in use
```
let's delete it

gemdev111 committed:
removed unused functions
